src/services/issue_analysis_service.py

```diff
@@ -0,0 +1,347 @@
+"""
+IssueAnalysisService - Responsible for issue analysis workflows.
+Handles filtering known issues, analyzing issues, and generating recommendations.
+"""
+
+import os
+import logging
+from typing import Tuple
+from langchain_core.prompts import ChatPromptTemplate, SystemMessagePromptTemplate, HumanMessagePromptTemplate
+from langchain_core.runnables import RunnablePassthrough, RunnableLambda
+from langchain_core.language_models.chat_models import BaseChatModel
+from langchain_community.vectorstores import FAISS
+from tenacity import retry, stop_after_attempt, wait_fixed, retry_if_exception_type
+
+from common.constants import FALLBACK_JUSTIFICATION_MESSAGE, RED_ERROR_FOR_LLM_REQUEST
+from dto.Issue import Issue
+from dto.ResponseStructures import FilterResponse, JudgeLLMResponse, JustificationsSummary, RecommendationsResponse, EvaluationResponse
+from dto.LLMResponse import AnalysisResponse
+from Utils.file_utils import read_answer_template_file
+from Utils.llm_utils import robust_structured_output
+from .vector_store_service import VectorStoreService
+
+logger = logging.getLogger(__name__)
+
+
+def _format_context_from_response(resp):
+    """Helper function to format context from vector search response"""
+    context_list = []
+    for doc in resp:
+        context_list.append({
+            "false_positive_error_trace": doc.page_content,
+            "reason_marked_false_positive": doc.metadata['reason_of_false_positive']
+        })
+    return context_list
+
+
+class IssueAnalysisService:
+    """Service for analyzing issues and determining false positives"""
+    
+    def __init__(self, config, vector_service: VectorStoreService):
+        self.config = config
+        self.vector_service = vector_service
+        self.max_retry_limit = 3
+        
+        # Store prompt templates from config
+        self.analysis_system_prompt = config.ANALYSIS_SYSTEM_PROMPT
+        self.analysis_human_prompt = config.ANALYSIS_HUMAN_PROMPT
+        self.filter_system_prompt = config.FILTER_SYSTEM_PROMPT
+        self.filter_human_prompt = config.FILTER_HUMAN_PROMPT
+        self.recommendations_prompt = config.RECOMMENDATIONS_PROMPT
+        self.justification_summary_system_prompt = config.JUSTIFICATION_SUMMARY_SYSTEM_PROMPT
+        self.justification_summary_human_prompt = config.JUSTIFICATION_SUMMARY_HUMAN_PROMPT
+        self.evaluation_prompt = config.EVALUATION_PROMPT
+        self.similarity_error_threshold = config.SIMILARITY_ERROR_THRESHOLD
```

Comment:
I would remove this and directly use it from the config object. Because by this point, we have already performed the validations.

Reply:
ok replacing all of them with `self.config.SPECIFIC_PROMPT_ACCORDINGLY`